### PR TITLE
chore: Push simply by `git push`

### PR DESCRIPTION
**Pull request by szepeviktor**
No need for an action.

This is a test repository
https://github.com/szepeviktor/git-push-from-github-actions/blob/master/.github/workflows/git-push.yml#L4

### CODE REVIEW

File: .github/workflows/go.yml [hunk 1]
```diff
@@ -60,6 +60,8 @@ jobs:
     needs: download
     runs-on: ubuntu-latest
     name: Update coverage badge
+    permissions:
+      contents: write
```
Comment (jd4235) [inline .github/workflows/go.yml:64]: What is the rationale to add content write permissions in the Github workflow?

Comment (szepeviktor) [reply .github/workflows/go.yml:64]: This adds write permission to the default GitHub token `github.token`.

Comment (szepeviktor) [reply .github/workflows/go.yml:66]: ...which may be the default. Being explicit is better than implicit.

Comment (szepeviktor) [reply .github/workflows/go.yml:66]: Example
image

Current value
https://github.com/Boeing/config-file-validator/settings/actions

Comment (szepeviktor) [reply .github/workflows/go.yml:66]: Here is the solution. Repository-wide write permissin 🤯 
https://github.com/ad-m/github-push-action/issues/189#issuecomment-1816532638

Comment (kehoecj) [reply .github/workflows/go.yml:66]: Thanks for the analysis @szepeviktor - that's really helpful

**Review by kehoecj — CHANGES_REQUESTED** [review]

File: .github/workflows/go.yml (hunk 1 above)
Comment (szepeviktor) [reply .github/workflows/go.yml:66]: Second one is also done.
I use `git diff` in many repositories. Could you tell me how it misbehaves?

Comment (kehoecj) [reply .github/workflows/go.yml:66]: I believe the issue is that `git diff --quiet` exits 1 when there is a changed file whereas `git status --porcelain` exits 0. For example here is the output of both commands after I made a manual change to `index.md`:

```bash
>> vim index.md 
>> git diff --quiet
>> echo $?
1
>> git status --porcelain
 M index.md
>> echo $?
0
```

Comment (szepeviktor) [reply .github/workflows/go.yml:66]: I see!
Actually `git status` does not changes its exit status on file change.
Please consider using `git diff --quiet`.

**Review by kehoecj — APPROVED** [review]

**Comment by kehoecj** [comment]
I tested all changes in this PR and validated them on a fork. Thanks @szepeviktor for the PR!

**Pull request merged**